### PR TITLE
Extend gadgets to print container & pod information

### PR DESCRIPTION
**Pull request by mauriciovasquezbernal**
This PR extends the gadgets to print information about the pod & container where the event happened. 

The gadget tracer manager creates an eBPF map where it saves all the details of the container. The different gadgets then perform a lookup in this map to get the information about the container and print it. This PR also changes how the node name is printed, now the full name (instead of just an integer is printed). 

Example of the output for the execsnoop gadget
```
$ ./kubectl-gadget-linux-amd64 execsnoop 
NODE             NAMESPACE        PODNAME          CONTAINERNAME   PCOMM            PID    PPID   RET ARGS
127.0.0.1        default          pod2             pod2            ping             227616 53305    0 /bin/ping 8.8.8.8
127.0.0.1        default          pod2             pod2            cat              227719 53305    0 /bin/cat /dev/null
^C
Terminating...
```

Example of output for the opensnoop gadget:
```
$ ./kubectl-gadget-linux-amd64 opensnoop 
NODE             NAMESPACE        PODNAME          CONTAINERNAME   PID    COMM               FD ERR PATH
127.0.0.1        default          pod              pod             380870 cat                 3   0 /etc/passwd
127.0.0.1        default          pod              pod             36996  sh                  3   0 /etc/passwd
127.0.0.1        default          pod2             pod2            380520 sh                  3   0 /
127.0.0.1        default          pod2             pod2            380520 sh                  3   0 /etc/
127.0.0.1        default          pod2             pod2            380520 sh                  3   0 /root/.ash_history
127.0.0.1        default          pod2             pod2            380961 cat                -1   2 /etc/host
127.0.0.1        default          pod2             pod2            380520 sh                  3   0 /etc/passwd
^C
Terminating...
```

Changes in bcc are available here -> https://github.com/kinvolk/bcc/pull/1

TODO:
- [x] Change BCC container image tag after https://github.com/kinvolk/bcc/pull/1 is merged

### CODE REVIEW

**Comment by alban** [comment]
Per commit review:

* Gadget Tracer Manager: export container details in BPF map 
  * I have made changes #189, so I suggest to either merge #189 first, or let me know if I should do a separate PR just for this.
* cmd/kubectl-gadget: do not append node info to gadget's output 
  * LGTM
* Update bcc
  * To be updated once https://github.com/kinvolk/bcc/pull/1 is merged.
  * I think the comment should be updated to say we use the branch "gadget"
* cmd/kubectl-gadget: add --containersmap parameter to gadgets 
  * LGTM

On the outputs: it seems some gadgets show `PCOMM` and others show `COMM`. Should it be standardised? Also, sometime the first column is "PCOMM", and sometime it is "NODE".

**Comment by mauriciovasquezbernal** [comment]
> let me know if I should do a separate PR just for this.

@alban that would be great to avoid creating a dependency between these two PRs.

**Comment by mauriciovasquezbernal** [comment]
> On the outputs: it seems some gadgets show PCOMM and others show COMM. Should it be standardised? 

I think it's an upstream issue. 

> Also, sometime the first column is "PCOMM", and sometime it is "NODE".

Do you have an example of which gadgets are doing it that way? I tried them and I always get `NODE` as the first one.

**Comment by alban** [comment]
> Do you have an example of which gadgets are doing it that way? I tried them and I always get NODE as the first one.

I was just looking at the description of this PR, I haven't tried it. The code looks ok :)

**Comment by mauriciovasquezbernal** [comment]
> I was just looking at the description of this PR, I haven't tried it. The code looks ok :)

That's outdated! I'll update it now!

**Comment by mauriciovasquezbernal** [comment]
@alban I handled your comments and rebased now that https://github.com/kinvolk/inspektor-gadget/pull/211 is merged.

File: gadget.Dockerfile [hunk 1]
```diff
@@ -33,10 +33,10 @@ FROM docker.io/kinvolk/traceloop:202006050210553a5730 as traceloop
 # Main gadget image
 
 # BCC built from:
-# https://github.com/kinvolk/bcc/commit/321c9c979889abce48d0844b3d539ec9a01e6f3c
+# https://github.com/kinvolk/bcc/commit/c5ad55e0067b4d87e503f80367066c5ae1d7acac
```
Comment (alban) [inline gadget.Dockerfile:36]: > the comment should be updated to say we use the branch "gadget"

I meant this comment in addition to the commitmsg:
 
```suggestion
# BCC built from the gadget branch at:
#
# https://github.com/kinvolk/bcc/commit/c5ad55e0067b4d87e503f80367066c5ae1d7acac
```

**Review by alban — APPROVED** [review]
One comment to change, then LGTM

File: gadget.Dockerfile (hunk 1 above)
Comment (mauriciovasquezbernal) [reply gadget.Dockerfile]: Ah, I thought you were referring to the commit message. I updated the comment in the docker file as well.

**Review by alban — APPROVED** [review]
LGTM

**Pull request merged**